internal/store/interfaces.go

```diff
@@ -58,3 +58,27 @@ type Store interface {
 	UpsertRoute(ctx context.Context, route Route, updateConditionFn func(current Route) bool) error
 	Sync(ctx context.Context) error
 }
+
+// Backend is used for persisting and querying gateways and routes
+type Backend interface {
+	GetGateway(ctx context.Context, id core.GatewayID) ([]byte, error)
+	ListGateways(ctx context.Context) ([][]byte, error)
+	DeleteGateway(ctx context.Context, id core.GatewayID) error
+	UpsertGateways(ctx context.Context, gateways ...GatewayRecord) error
+	GetRoute(ctx context.Context, id string) ([]byte, error)
+	ListRoutes(ctx context.Context) ([][]byte, error)
+	DeleteRoute(ctx context.Context, id string) error
+	UpsertRoutes(ctx context.Context, routes ...RouteRecord) error
+}
+
+// GatewayRecord represents a serialized Gateway
+type GatewayRecord struct {
+	ID   core.GatewayID
+	Data []byte
+}
+
+// RouteRecord represents a serialized Route
+type RouteRecord struct {
+	ID   string
```

Comment:
Know this code is exactly what we currently leverage, but just calling out here that after the refactor is done we might want to switch this to be similar to `GatewayID` -- a compound struct with namespace/name info. Nothing to do for now though.